Tasks/DownloadBuildArtifacts/main.ts

```diff
@@ -0,0 +1,139 @@
+
+        var endpointUrl : string = tl.getVariable("System.TeamFoundationCollectionUri");
+        var accessToken : string = tl.getEndpointAuthorizationParameter('SYSTEMVSSCONNECTION', 'AccessToken', false);
+        var credentialHandler : IRequestHandler = getHandlerFromToken(accessToken);
+        var webApi: WebApi = new WebApi(endpointUrl, credentialHandler);
+        var debugMode : string = tl.getVariable('System.Debug');
+        var isVerbose : boolean = debugMode ? debugMode.toLowerCase() != 'false' : false;
+        var parallelLimit : number = +tl.getVariable("release.artifact.download.parallellimit");
+
+        var templatePath: string = path.join(__dirname, 'vsts.handlebars.txt');
+        var buildApi : IBuildApi = webApi.getBuildApi();
+        var artifacts = [];
+        var itemPattern : string = '**';
+
+        // verfiy that buildId belongs to the definition selected
+        if (definitionId) {
+            var builds = await buildApi.getBuilds(projectId, [parseInt(definitionId)]).catch((reason) => {
+                reject(reason);
+            });
+
+            if (builds) {
+                var buildIds = builds.map((value, index) => {
+                    return value.id;
+                });
+
+                if (buildIds.indexOf(buildId) == -1) {
+                    reject(tl.loc("BuildIdBuildDefinitionMismatch", buildId, definitionId));
+                }
+            }
+            else {
+                reject(tl.loc("NoBuildsFound", definitionId));
+            }
+        }
+
+        // populate itempattern and artifacts based on downloadType
+        if (downloadType === 'single') {
+            var artifactName = tl.getInput("artifactName");
+            var artifact = await buildApi.getArtifact(buildId, artifactName, projectId).catch((reason) => {
+                reject(reason);
+            });
+            artifacts.push(artifact);
+            itemPattern = artifactName + '/**';
+        }
+        else {
+            var buildArtifacts = await buildApi.getArtifacts(buildId, projectId).catch((reason) => {
+                reject(reason);
+            });
+
+            console.log(tl.loc("LinkedArtifactCount", buildArtifacts.length));
+            artifacts = artifacts.concat(buildArtifacts);
+            itemPattern = tl.getInput("itemPattern", false) || '**';
+        }
+
+        var downloadPromises: Array<Promise<any>> = [];
+
+        artifacts.forEach(async function (artifact, index, artifacts) {
+            let downloaderOptions = new engine.ArtifactEngineOptions();
+            downloaderOptions.itemPattern = itemPattern;
+            downloaderOptions.verbose = isVerbose;
+
+            if(parallelLimit){
+                downloaderOptions.parallelProcessingLimit = parallelLimit;
+            }
+
+            if (artifact.resource.type.toLowerCase() === "container") {
+                let downloader = new engine.ArtifactEngine();
```

Comment:
the downloader emits a lot of strings that are not localized...

Reply:
we don't have support for localization in vsts-rm-extensions currently. It is on our backlog.

Reply:
What is the schedule for getting this done?  Localization is pretty important for on-prem releases.

Reply:
Agree. We plan to take this up in next wave. But definitely before the task ships on-prem.